src/mode/graphics.rs

```diff
@@ -116,13 +130,65 @@ where
         self.properties
             .set_draw_area((0, 0), (display_width, display_height))?;
 
+        self.min_x = 255;
+        self.max_x = 0;
+        self.min_y = 255;
+        self.max_y = 0;
+
         match display_size {
             DisplaySize::Display128x64 => self.properties.draw(&self.buffer),
             DisplaySize::Display128x32 => self.properties.draw(&self.buffer[0..512]),
             DisplaySize::Display96x16 => self.properties.draw(&self.buffer[0..192]),
         }
     }
 
+    /// Write out data to a display.
+    /// 
+    /// This is typically faster than a regular flush since it only updates the parts of the
+    /// display that have changed since the last flush. 
+    /// 
+    /// This is slower than a regular flush when the size of the updated area approaches the full
+    /// size of the display, so in that case this function simply calls flush.
+    pub fn fast_flush(&mut self) -> Result<(), DI::Error> {
+        if self.max_x < self.min_x || self.max_y < self.min_y {
+            return self.flush();
+        }
+
+        let display_size = self.properties.get_size();
+        let (width, height) = display_size.dimensions();
+        let width = width as usize;
+
+        // Determine which bytes need to be sent
+        let disp_min_x = self.min_x;
+        let disp_min_y = self.min_y;
+
+        let disp_max_x = if self.max_x + 1 > width as u8 {
```

Comment:
This `if`/`else` could be replaced with

```suggestion
        let disp_max_x = (self.max_x + 1).min(width as u8)
```

Reply:
I was not familiar with `.min()`, that makes things much nicer.